index.js

```diff
@@ -221,15 +223,12 @@ class ResultStore {
         }
 
         // logging results
-        if (obj.emit) this.conn.loginfo(plugin, result.human);  // by request
+        if (obj.emit  && result.human.trim().length > 0) this.conn.loginfo(plugin, result.human);  // by request
         if (obj.err) {
             // Handle error objects by logging the message
-            if (util.isError(obj.err)) {
-                this.conn.logerror(plugin, obj.err.message);
-            }
-            else {
-                this.conn.logerror(plugin, obj.err);
-            }
+            const level = obj.emit_log_level ? obj.emit_log_level : 'error';
+            const logobj = util.isError(obj.err) ? obj.err.message : obj.err;
```

Comment:
The name `logobj` implies that the contents are an object and that's not the case. Lets call this 'logmsg' or even just 'msg', at your discretion.